In [1]:
import pandas as pd
import statistics as stats
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df= pd.read_csv("spambase.csv")

In [3]:
dataset = pd.read_csv

In [4]:
df.head()

Unnamed: 0  word_freq_make  word_freq_address  word_freq_all  word_freq_3d  \
0           0            0.00               0.64           0.64           0.0   
1           1            0.21               0.28           0.50           0.0   
2           2            0.06               0.00           0.71           0.0   
3           3            0.00               0.00           0.00           0.0   
4           4            0.00               0.00           0.00           0.0   

  word_freq_our  word_freq_over  word_freq_remove  word_freq_internet  \
0          0.32            0.00              0.00                0.00   
1          0.14            0.28              0.21                0.07   
2          1.23            0.19              0.19                0.12   
3          0.63            0.00              0.31                0.63   
4          0.63            0.00              0.31                0.63   

   word_freq_order  ...  char_freq_;  char_freq_(  char_freq_[  char_freq_!  \
0             0.00  ...         0.00        0.000          0.0        0.778   
1             0.00  ...         0.00        0.132          0.0        0.372   
2             0.64  ...         0.01        0.143          0.0        0.276   
3             0.31  ...         0.00        0.137          0.0        0.137   
4             0.31  ...         0.00        0.135          0.0        0.135   

   char_freq_$  char_freq_#  capital_run_length_average  \
0        0.000        0.000                       3.756   
1        0.180        0.048                       5.114   
2        0.184        0.010                       9.821   
3        0.000        0.000                       3.537   
4        0.000        0.000                       3.537   

   capital_run_length_longest  capital_run_length_total  is_spam  
0                          61                       278     True  
1                         101                      1028     True  
2                         485                      2259     True  
3                          40                       191     True  
4                          40                       191     True  

[5 rows x 59 columns]

In [5]:
df.shape

(4601, 59)

 Dataset contains 4601 observations (rows) and 59 features (columns). The number of attributes is 58 (57 continuous and 1 nominal class label).
 Additional information obtained from "spambase_documentation" shows that spam constitutes 39.4% (1813 emails), while 2788 email are not spam (60.6%). 
 The attributes are continuous (attributes of type word_freq_WORD, attributes of type char_freq_CHAR and attributes of type capital_run_lenght) and also categorical (last feature -indicator "is_spam" which is a binary attribute)

This data shows the characteristics of emails based on their attributes Iword_freq,char_freq and capital_run_lenght). This characteristic is very important for identifying spam ( due to repeated charcters of words or frequent use of capital letters we can ditinguish spam from regular email).
We can also compare the diffrences between spam and non-spam. 